drivers/include/periph/wdt.h

```diff
@@ -230,6 +230,23 @@ extern "C" {
 #define WDT_HAS_INIT              (0)
 #endif
```

Comment:
Please add some doc promoting the `auto_init_wdt_thread` module here and warn users that this will only be able to detect when low priority threads starve. We don't want users to assume this can replace a sophisticated watchdog implementation that actually can reason about the state of the firmware due to some baked in knowledge about the application logic. After all, this cannot magically detect all kinds of issues while staying so generic that it can just be used via selecting a module.

Reply:
I'm actually beginning to wonder if the watchdog thread can be useful at all - it will only detect high load / lifelock situations.

The early watchdog on the other hand is certainly useful if something hangs during init.

Reply:
Thinking about it, maybe an event thread would be an even better choice.

This would also enforce the 'no long tasks in the event thread' assertion.